synapse/event_auth.py

```diff
@@ -80,6 +80,31 @@ def validate_event_for_room_version(event: "EventBase") -> None:
     Raises:
         SynapseError if there is a problem with the event
     """
+    # Reject events with stringy power levels if required by room version
```

Comment:
I added the check here because the spec seems to indicate that the power level check failure should be "rejected under the authorization rules " and this was an auth check function relating to room versions, so it seemed the appropriate place. If it's not I am happy to move it.

Reply:
Can we move this to `_check_power_levels`, where we apply the rest of the auth rules under item 10. in the spec (https://spec.matrix.org/latest/rooms/v10/#authorization-rules) ?

Reply:
Yes, done in 808dd94.